experimental/rest_client.go

```diff
@@ -0,0 +1,90 @@
+package experimental
+
+import (
+	"context"
+	"crypto/tls"
+	"errors"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/url"
+	"path"
+	"time"
+)
+
+// Client interface
+type Client interface {
+	Fetch(ctx context.Context, uriPath, uriQuery string) ([]byte, error)
+	Get(ctx context.Context, uriPath, uriQuery string) (*http.Response, error)
+}
+
+type restClient struct {
+	url     string
+	headers map[string]string
+	client  *http.Client
+}
+
+// NewRestClient ...
+func NewRestClient(url string, headers map[string]string) Client {
+	return &restClient{
+		url:     url,
+		headers: headers,
+		client: &http.Client{
```

Comment:
I changed your original so it does not use a global single http client for everything, rather it create a new client for each `NewRestClient`

I wonder if we should take the DataSourceInstanceSettings as a conig object and support the standard TLS settings too?